mona_openai/loggers/logger.py

```diff
@@ -0,0 +1,43 @@
+import abc
+
+
+class Logger(metaclass=abc.ABCMeta):
+    """
+    An abstract class/interface for logging messages containing OpenAI calls
+    analysis data.
+    """
+
+    def start_monitoring(self, openai_class_name):
+        """
+        This function will be called once this logger is used for wrapping
+        an OpenAI class.
+        Child classes may choose to use this hook in order to run some logic
+        preliminary to actual logging.
+        """
+        pass
+
+    @abc.abstractmethod
+    def log(self, message: dict, context_id=None, export_timestamp=None):
+        """
+        Every child class must implement this basic function which gets a
+        dictionary to be logged.
```

Comment:
No this is a message to be logged. We might want to set a rule over here that it must be JSON serializable - but I decided to avoid any assumptions at this stage.